aws_iam_policies/aws_iam_role_external_permission.yml

```diff
@@ -0,0 +1,327 @@
+              }
+            }
+          ],
+          "Version": "2012-10-17"
+        },
+        "CreateDate": "2018-01-01T19:00:00+00:00",
+        "InlinePolicies": {
+          "TestPolicy1": "{\"Version\": \"2012-10-17\",\"Statement\": [{ \"Sid\": \"TestSid\",\"Effect\": \"Allow\",\"Action\": [\"test:Permission1\", \"test:Permission1\"],\"Resource\": \"*\"}]}",
+          "TestPolicy2": "{\"Version\": \"2012-10-17\",\"Statement\": [{ \"Sid\": \"TestSid\",\"Effect\": \"Allow\",\"Action\": [\"test:Permission1\", \"test:Permission2\"],\"Resource\": \"*\"}]}"
+        },
+        "ManagedPolicyNames": [
+          "CONFIGURATION_REQUIRED_Policy_must_actually_exist_in_Principal_acct_and_NOT_have_specified_permission",
+          "Alternate_method_is_create_mock_see_instructions_in_policy"
+        ],
+        "IsUnitTest": true,
+        "DoesNotHavePermission": true,
+        "Path": "/",
+        "RoleId": "ROLEID",
+        "RoleName": "SampleRole"
+      }
+  -
+    Name: Managed Policy Not Found Internal Account
+    ExpectedResult: true
+    Resource:
+      {
+        "AccountId": "123456789012",
+        "Arn": "arn:aws:iam::123456789012:role/SampleRole",
+        "AssumeRolePolicyDocument": {
+          "Statement": [
+            {
+              "Action": "sts:AssumeRole",
+              "Condition": {},
+              "Effect": "Allow",
+              "Principal": {
+                "AWS": "arn:aws:iam::123456789012:role/CONFIGURATION_REQUIRED_set_AccountId_to_any_org_acct"
+              }
+            }
+          ],
+          "Version": "2012-10-17"
+        },
+        "CreateDate": "2018-01-01T19:00:00+00:00",
+        "CONFIGURATION_REQUIRED": "set Action: <specified permission> in TestPermPolicy",
+        "InlinePolicies": null,
+        "ManagedPolicyNames": [
+            "PolicyDoesNotExist"
+          ],
+        "Path": "/",
+        "RoleId": "ROLEID",
+        "RoleName": "SampleRole"
+      }
+  -
+    Name: Null Managed Policy
+    ExpectedResult: true
+    Resource:
+      {
+        "AccountId": "123456789012",
+        "Arn": "arn:aws:iam::123456789012:role/SampleRole",
+        "AssumeRolePolicyDocument": {
+          "Statement": [
+            {
+              "Action": "sts:AssumeRole",
+              "Condition": {},
+              "Effect": "Allow",
+              "Principal": {
+                "AWS": "arn:aws:iam::123456789012:role/CONFIGURATION_REQUIRED_set_AccountId_to_any_org_acct"
+              }
+            }
+          ],
+          "Version": "2012-10-17"
+        },
+        "CreateDate": "2018-01-01T19:00:00+00:00",
+        "CONFIGURATION_REQUIRED": "set Action: <specified permission> in TestPermPolicy",
+        "InlinePolicies": null,
+        "ManagedPolicyNames": null,
+        "Path": "/",
+        "RoleId": "ROLEID",
+        "RoleName": "SampleRole"
+      }
+  -
+    Name: Non-AWS Trust Principal
+    ExpectedResult: true
+    Resource:
+      {
+	    "AccountId": "123456789012",
```

Comment:
I've never seen these red bars - is it tabs whereas the rest of the whitespaces are spaces? If so, should we convert to be consistent?

Reply:
Yes and yes.